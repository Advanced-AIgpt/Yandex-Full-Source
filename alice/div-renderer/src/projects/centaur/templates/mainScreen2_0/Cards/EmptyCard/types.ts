import { IAbstractCardProps } from '../types';

export interface IEmptyCardProps extends IAbstractCardProps {
    type: 'empty';
}
