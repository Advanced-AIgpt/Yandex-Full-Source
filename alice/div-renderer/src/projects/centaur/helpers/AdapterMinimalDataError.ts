export class AdapterMinimalDataError extends Error {
    constructor(missedData: string[], template: string) {
        super(`Missed data (${missedData.join(', ')}) in template ${template}`);
        this.name = 'AdapterMinimalDataError';
    }
}
