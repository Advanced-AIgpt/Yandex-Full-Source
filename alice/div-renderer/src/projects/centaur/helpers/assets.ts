export function getS3Asset(path: string) {
    return `https://dialogs.s3.yandex.net/smart_displays/${path}`;
}

export function getStaticS3Asset(path: string) {
    return `https://yastatic.net/s3/dialogs/smart_displays/${path}`;
}
