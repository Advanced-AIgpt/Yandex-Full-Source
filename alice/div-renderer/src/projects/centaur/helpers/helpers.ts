/* eslint-disable camelcase */
import { sortedIndexBy } from 'lodash';
import { format, isSameDay } from 'date-fns';
import { ru } from 'date-fns/locale';
import { Div, IDivData, TextBlock } from 'divcard2';
import { IRequestState } from '../../../common/types/common';
import { getTemplates, templatesRegistry } from '../../../registries/common';
import { centaurTemplatesClass } from '../index';
import { Palette } from './palette';

export { default as pluralize } from '@yandex-int/i18n/lib/plural/ru';
type TextProps = Pick<ConstructorParameters<typeof TextBlock>[0], 'text_color'|'font_size'|'font_weight'|'line_height'>;

export const textBreakpoint =
    (...breakpoints: [number, TextProps][]) =>
        (text: string) => {
            const idx = Math.min(sortedIndexBy(breakpoints, [text.length, {}], 0), breakpoints.length - 1);
            return { text, ...breakpoints[idx][1] };
        };

export const formatDate = (d: Date | string | number, f: string) => format(new Date(d), f, { locale: ru });

export const textCommand = (c: string) => `dialog://text_command?query=${encodeURI(c)}`;

export const updatedAt = (d: Date) => {
    const now = new Date();
    return `Обновлен ${isSameDay(now, d) ? 'сегодня' : formatDate(d, 'dd.MM')}`;
};

export const insertBetween = <I extends unknown, S extends unknown>(itemList: I[], separator: S) => {
    return itemList.reduce<(I | S)[]>((result, item, index) => {
        if (index !== 0) {
            result.push(separator);
        }

        result.push(item);

        return result;
    }, []);
};

export interface ITemplateCard {
    templates: {
        [name: string]: Div;
    };
    palette?: Palette;
    card: IDivData;
}

// Дива верхнего уровня. Замена для new `TemplateCard()`
export const TopLevelCard = (card: Omit<IDivData, 'variable_triggers' | 'variables'>, requestState: IRequestState, palette?: Palette) => {
    const templates = Object.assign(
        getTemplates(requestState.localTemplates, templatesRegistry),
        centaurTemplatesClass.getUsedTemplates(requestState),
        requestState.res.localTemplates.getAll(),
    );

    return {
        templates,
        palette,
        card: {
            ...card,
            variable_triggers: requestState.variableTriggers.getAll(),
            variables: [...requestState.variables],
        } as IDivData,
    };
};
