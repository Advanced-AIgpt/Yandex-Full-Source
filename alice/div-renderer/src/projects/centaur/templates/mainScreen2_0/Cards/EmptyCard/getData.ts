import { IEmptyCardProps } from './types';
import { ICardDataAdapter } from '../types';

export const getEmptyCardData: ICardDataAdapter<IEmptyCardProps> = function getEmptyCardData(card, requestState) {
    if (card && card.VacantCardData) {
        return {
            type: 'empty',
            requestState,
        };
    }
    return null;
};
