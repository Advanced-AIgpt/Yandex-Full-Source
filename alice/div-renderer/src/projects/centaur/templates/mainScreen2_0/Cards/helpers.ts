export function getCardMainScreenId({ rowIndex, colIndex }: { rowIndex?: number; colIndex?: number }) {
    return `main_screen_card_${rowIndex || 0}_${colIndex || 0}`;
}
