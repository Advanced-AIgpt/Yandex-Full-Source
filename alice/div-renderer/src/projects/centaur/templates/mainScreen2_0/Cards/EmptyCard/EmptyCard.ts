import { ContainerBlock, Div, DivStateBlock, FixedSize, ImageBlock, MatchParentSize, TextBlock } from 'divcard2';
import { compact } from 'lodash';
import { BasicCard } from '../BasicCard';
import { title32m } from '../../../../style/Text/Text';
import { colorWhiteOpacity50 } from '../../../../style/constants';
import { getS3Asset } from '../../../../helpers/assets';
import { IEmptyCardProps } from './types';
import { LoaderDiv } from '../../../../components/Loader/Loader';
import { setStateAction } from '../../../../../../common/actions/div';
import { getCardMainScreenId } from '../helpers';

const ADD_WIDGET_CARD = 'add_widget_loading';
const ADD_WIDGET_CARD_ON = 'on';
const ADD_WIDGET_CARD_OFF = 'off';

export default function EmptyCard({ actions, longtap_actions, rowIndex, colIndex } : IEmptyCardProps): Div {
    const divId = `${ADD_WIDGET_CARD}_${rowIndex ?? 0}_${colIndex ?? 0}`;

    return BasicCard({
        id: getCardMainScreenId({ colIndex, rowIndex }),
        height: new MatchParentSize(),
        width: new MatchParentSize(),
        items: [
            new DivStateBlock({
                div_id: divId,
                default_state_id: ADD_WIDGET_CARD_OFF,
                states: [
                    {
                        state_id: ADD_WIDGET_CARD_OFF,
                        div: new ContainerBlock({
                            height: new MatchParentSize(),
                            width: new MatchParentSize(),
                            actions: actions ? compact([
                                {
                                    log_id: 'mainscreen.emptycard.change.loading.state',
                                    url: setStateAction([
                                        '0',
                                        divId,
                                        ADD_WIDGET_CARD_ON,
                                    ]),
                                },
                                ...actions,
                            ]) : [],
                            longtap_actions,
                            items: [
                                new ImageBlock({
                                    height: new FixedSize({ value: 88 }),
                                    width: new FixedSize({ value: 88 }),
                                    margins: {
                                        top: 24,
                                        bottom: 24,
                                    },
                                    image_url: getS3Asset('icons/plus_in_circle.png'),
                                    alignment_horizontal: 'center',
                                }),
                                new TextBlock({
                                    ...title32m,
                                    text_color: colorWhiteOpacity50,
                                    width: new MatchParentSize(),
                                    height: new MatchParentSize(),
                                    text_alignment_horizontal: 'center',
                                    text: 'Добавить\nвиджет',
                                }),
                            ],
                        }),
                    },
                    {
                        state_id: ADD_WIDGET_CARD_ON,
                        div: new ContainerBlock({
                            height: new MatchParentSize(),
                            width: new MatchParentSize(),
                            actions: [
                                {
                                    log_id: 'mainscreen.emptycard.change.loading.state',
                                    url: setStateAction([
                                        '0',
                                        divId,
                                        ADD_WIDGET_CARD_OFF,
                                    ]),
                                },
                            ],
                            items: [
                                LoaderDiv(),
                            ],
                        }),
                    },
                ],
            }),
        ],
    });
}
