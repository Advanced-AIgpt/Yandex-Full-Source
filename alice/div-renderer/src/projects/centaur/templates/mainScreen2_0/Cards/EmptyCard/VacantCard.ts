import { Div, FixedSize, ImageBlock, MatchParentSize, TextBlock } from 'divcard2';
import { BasicCard } from '../BasicCard';
import { title32m } from '../../../../style/Text/Text';
import { colorWhiteOpacity50 } from '../../../../style/constants';
import { getS3Asset } from '../../../../helpers/assets';
import { IEmptyCardProps } from './types';
import { getCardMainScreenId } from '../helpers';

export default function VacantCard({ actions, longtap_actions, rowIndex, colIndex } : IEmptyCardProps): Div {
    return BasicCard({
        id: getCardMainScreenId({ rowIndex, colIndex }),
        height: new MatchParentSize(),
        width: new MatchParentSize(),
        actions,
        longtap_actions,
        items: [
            new ImageBlock({
                height: new FixedSize({ value: 88 }),
                width: new FixedSize({ value: 88 }),
                margins: {
                    top: 24,
                    bottom: 24,
                },
                image_url: getS3Asset('icons/plus_in_circle.png'),
                alignment_horizontal: 'center',
            }),
            new TextBlock({
                ...title32m,
                text_color: colorWhiteOpacity50,
                width: new MatchParentSize(),
                height: new MatchParentSize(),
                text_alignment_horizontal: 'center',
                text: 'Убрать виджет',
            }),
        ],
    });
}
