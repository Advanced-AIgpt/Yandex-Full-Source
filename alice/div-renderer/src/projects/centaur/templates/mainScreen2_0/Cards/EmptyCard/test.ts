import EmptyCard from './EmptyCard';
import { AnonymizeDataForSnapshot } from '../../../../../../common/helpers/dev';

jest.mock('../../../../../../common/logger');

describe('Main screen empty card', () => {
    it('should match snapshot', () => {
        expect(AnonymizeDataForSnapshot(EmptyCard({
            type: 'empty',
        }))).toMatchSnapshot();
    });
});
