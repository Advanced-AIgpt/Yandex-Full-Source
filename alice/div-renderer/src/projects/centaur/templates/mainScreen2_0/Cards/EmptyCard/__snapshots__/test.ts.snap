// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main screen empty card should match snapshot 1`] = `
Object {
  "background": Array [
    Object {
      "color": "#1aeaf6ec",
      "type": "solid",
    },
  ],
  "border": Object {
    "corner_radius": 28,
  },
  "height": Object {
    "type": "match_parent",
  },
  "id": "main_screen_card_0_0",
  "items": Array [
    Object {
      "default_state_id": "off",
      "div_id": "add_widget_loading_0_0",
      "states": Array [
        Object {
          "div": Object {
            "actions": Array [],
            "height": Object {
              "type": "match_parent",
            },
            "items": Array [
              Object {
                "alignment_horizontal": "center",
                "height": Object {
                  "type": "fixed",
                  "value": 88,
                },
                "image_url": "https://dialogs.s3.yandex.net/smart_displays/icons/plus_in_circle.png",
                "margins": Object {
                  "bottom": 24,
                  "top": 24,
                },
                "type": "image",
                "width": Object {
                  "type": "fixed",
                  "value": 88,
                },
              },
              Object {
                "font_size": 32,
                "font_weight": "medium",
                "height": Object {
                  "type": "match_parent",
                },
                "line_height": 40,
                "text": "Добавить
виджет",
                "text_alignment_horizontal": "center",
                "text_color": "#80ffffff",
                "type": "text",
                "width": Object {
                  "type": "match_parent",
                },
              },
            ],
            "type": "container",
            "width": Object {
              "type": "match_parent",
            },
          },
          "state_id": "off",
        },
        Object {
          "div": Object {
            "actions": Array [
              Object {
                "log_id": "mainscreen.emptycard.change.loading.state",
                "url": "div-action://set_state?state_id=0/add_widget_loading_0_0/off&temporary=false",
              },
            ],
            "height": Object {
              "type": "match_parent",
            },
            "items": Array [
              Object {
                "alignment_horizontal": "center",
                "alignment_vertical": "center",
                "gif_url": "https://dialogs.s3.yandex.net/smart_displays/loaders/spinner.gif",
                "height": Object {
                  "type": "fixed",
                  "value": 250,
                },
                "type": "gif",
                "width": Object {
                  "type": "fixed",
                  "value": 250,
                },
              },
            ],
            "type": "container",
            "width": Object {
              "type": "match_parent",
            },
          },
          "state_id": "on",
        },
      ],
      "type": "state",
    },
  ],
  "paddings": Object {
    "bottom": 24,
    "left": 24,
    "right": 24,
    "top": 24,
  },
  "type": "container",
  "width": Object {
    "type": "match_parent",
  },
}
`;
